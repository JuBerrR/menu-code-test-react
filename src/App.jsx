import React from 'react';
import { render } from 'react-dom';
import Checkout from './components/checkout/Checkout';
import Menu from './components/menu/Menu';
import { MenuProvider } from './context/MenuContext';

import './App.scss';

const App = () => {
    return (
        <MenuProvider>
            <header className="header">Menu Test</header>
            <div className="app-container">
                <div className="menu-list">
                    <Menu />
                </div>
                <div className="checkout">
                    <Checkout />
                </div>
            </div>
        </MenuProvider>
    );
};

render(<App />, document.getElementById('root'));
// render(<App />, document.getElementById('root') || document.createElement('div')); // for testing

export default App;
