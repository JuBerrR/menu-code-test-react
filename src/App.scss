$color-blue: #0059ff;

.header {
    text-align: center;
    font-size: 5rem;
    margin: 8px 48px;
}

.app-container {
    display: flex;
    margin: 16px 20%;

    .menu-list {
        background-color: #f5e2f0b6;
        margin-right: 8px;
        width: 75%;
        min-width: 250px;
        text-align: center;
        border-radius: 4px;
    }

    .checkout {
        width: 25%;
        min-width: 100px;
    }
}
