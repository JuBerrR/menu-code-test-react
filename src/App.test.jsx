import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

describe('<App />', () => {
    it('Renders <App /> component correctly', () => {
        const app = render(<App />);

        expect(app).toMatchSnapshot();
    });
});
