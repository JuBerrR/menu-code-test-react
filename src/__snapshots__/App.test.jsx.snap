// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Renders <App /> component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        Menu Test
      </header>
      <div
        class="app-container"
      >
        <div
          class="menu-list"
        >
          <div>
            <div>
              <h2>
                Starters
              </h2>
              <div
                class="menu-items-container"
              >
                <span>
                  <button
                    class="dish"
                    type="button"
                  >
                    Soup
                    <p
                      class="dish-price"
                    >
                      3
                       Eur
                    </p>
                  </button>
                </span>
                <span>
                  <button
                    class="dish"
                    type="button"
                  >
                    Pâté
                    <p
                      class="dish-price"
                    >
                      5
                       Eur
                    </p>
                  </button>
                </span>
                <span>
                  <button
                    class="dish"
                    type="button"
                  >
                    Bruschetta
                    <p
                      class="dish-price"
                    >
                      4.5
                       Eur
                    </p>
                  </button>
                </span>
                <span>
                  <button
                    class="dish"
                    type="button"
                  >
                    Prawn cocktail
                    <p
                      class="dish-price"
                    >
                      6
                       Eur
                    </p>
                  </button>
                </span>
              </div>
            </div>
            <div>
              <h2>
                Mains
              </h2>
              <div
                class="menu-items-container"
              >
                <span>
                  <button
                    class="dish"
                    type="button"
                  >
                    Steak
                    <p
                      class="dish-price"
                    >
                      18
                       Eur
                    </p>
                  </button>
                </span>
                <span>
                  <button
                    class="dish"
                    type="button"
                  >
                    Meatballs
                    <p
                      class="dish-price"
                    >
                      11.5
                       Eur
                    </p>
                  </button>
                </span>
                <span>
                  <button
                    class="dish"
                    type="button"
                  >
                    Salmon fillet
                    <p
                      class="dish-price"
                    >
                      14
                       Eur
                    </p>
                  </button>
                </span>
                <span>
                  <button
                    class="dish"
                    type="button"
                  >
                    Vegetarian lasagna
                    <p
                      class="dish-price"
                    >
                      12
                       Eur
                    </p>
                  </button>
                </span>
              </div>
            </div>
            <div>
              <h2>
                Desserts
              </h2>
              <div
                class="menu-items-container"
              >
                <span>
                  <button
                    class="dish"
                    type="button"
                  >
                    Sticky toffee
                    <p
                      class="dish-price"
                    >
                      18
                       Eur
                    </p>
                  </button>
                </span>
                <span>
                  <button
                    class="dish"
                    type="button"
                  >
                    Tiramisu
                    <p
                      class="dish-price"
                    >
                      4.5
                       Eur
                    </p>
                  </button>
                </span>
                <span>
                  <button
                    class="dish"
                    type="button"
                  >
                    Cheesecake
                    <p
                      class="dish-price"
                    >
                      4
                       Eur
                    </p>
                  </button>
                </span>
                <span>
                  <button
                    class="dish"
                    type="button"
                  >
                    Ice cream
                    <p
                      class="dish-price"
                    >
                      3.5
                       Eur
                    </p>
                  </button>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="checkout"
        >
          <div
            class="checkout-container"
          >
            <header
              class="checkout-header"
            >
              <button
                class="select-diner-button left"
                data-testid="select-diner-button left"
                disabled=""
                type="button"
              >
                Diner 1
              </button>
              <button
                class="select-diner-button right"
                data-testid="select-diner-button right"
                type="button"
              >
                Diner 2
              </button>
            </header>
            <div
              class="checkout-items"
            />
            <div
              class="checkout-footer"
            >
              <div
                class="total-bill-label"
              />
              <button
                class="order-button"
                data-testid="order-button"
                disabled=""
                type="button"
              >
                Order
              </button>
              <div
                class="error-text"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      Menu Test
    </header>
    <div
      class="app-container"
    >
      <div
        class="menu-list"
      >
        <div>
          <div>
            <h2>
              Starters
            </h2>
            <div
              class="menu-items-container"
            >
              <span>
                <button
                  class="dish"
                  type="button"
                >
                  Soup
                  <p
                    class="dish-price"
                  >
                    3
                     Eur
                  </p>
                </button>
              </span>
              <span>
                <button
                  class="dish"
                  type="button"
                >
                  Pâté
                  <p
                    class="dish-price"
                  >
                    5
                     Eur
                  </p>
                </button>
              </span>
              <span>
                <button
                  class="dish"
                  type="button"
                >
                  Bruschetta
                  <p
                    class="dish-price"
                  >
                    4.5
                     Eur
                  </p>
                </button>
              </span>
              <span>
                <button
                  class="dish"
                  type="button"
                >
                  Prawn cocktail
                  <p
                    class="dish-price"
                  >
                    6
                     Eur
                  </p>
                </button>
              </span>
            </div>
          </div>
          <div>
            <h2>
              Mains
            </h2>
            <div
              class="menu-items-container"
            >
              <span>
                <button
                  class="dish"
                  type="button"
                >
                  Steak
                  <p
                    class="dish-price"
                  >
                    18
                     Eur
                  </p>
                </button>
              </span>
              <span>
                <button
                  class="dish"
                  type="button"
                >
                  Meatballs
                  <p
                    class="dish-price"
                  >
                    11.5
                     Eur
                  </p>
                </button>
              </span>
              <span>
                <button
                  class="dish"
                  type="button"
                >
                  Salmon fillet
                  <p
                    class="dish-price"
                  >
                    14
                     Eur
                  </p>
                </button>
              </span>
              <span>
                <button
                  class="dish"
                  type="button"
                >
                  Vegetarian lasagna
                  <p
                    class="dish-price"
                  >
                    12
                     Eur
                  </p>
                </button>
              </span>
            </div>
          </div>
          <div>
            <h2>
              Desserts
            </h2>
            <div
              class="menu-items-container"
            >
              <span>
                <button
                  class="dish"
                  type="button"
                >
                  Sticky toffee
                  <p
                    class="dish-price"
                  >
                    18
                     Eur
                  </p>
                </button>
              </span>
              <span>
                <button
                  class="dish"
                  type="button"
                >
                  Tiramisu
                  <p
                    class="dish-price"
                  >
                    4.5
                     Eur
                  </p>
                </button>
              </span>
              <span>
                <button
                  class="dish"
                  type="button"
                >
                  Cheesecake
                  <p
                    class="dish-price"
                  >
                    4
                     Eur
                  </p>
                </button>
              </span>
              <span>
                <button
                  class="dish"
                  type="button"
                >
                  Ice cream
                  <p
                    class="dish-price"
                  >
                    3.5
                     Eur
                  </p>
                </button>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="checkout"
      >
        <div
          class="checkout-container"
        >
          <header
            class="checkout-header"
          >
            <button
              class="select-diner-button left"
              data-testid="select-diner-button left"
              disabled=""
              type="button"
            >
              Diner 1
            </button>
            <button
              class="select-diner-button right"
              data-testid="select-diner-button right"
              type="button"
            >
              Diner 2
            </button>
          </header>
          <div
            class="checkout-items"
          />
          <div
            class="checkout-footer"
          >
            <div
              class="total-bill-label"
            />
            <button
              class="order-button"
              data-testid="order-button"
              disabled=""
              type="button"
            >
              Order
            </button>
            <div
              class="error-text"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
